src/permute.js

```diff
@@ -1,3 +1,45 @@
-export default function(source, keys) {
-  return Array.from(keys, key => source[key]);
+export default function (source, keys) {
+  if (keys === undefined) {
+    return permute(source);
+  } else {
+    return Array.from(keys, (key) => source[key]);
+  }
+}
+
+// Based on https://github.com/google/guava/blob/f4b3f611c4e49ecaded58dcb49262f55e56a3322/guava/src/com/google/common/collect/Collections2.java#L622-L683
+// Apache-2.0 License, Copyright (C) 2008 The Guava Authors.
```

Comment:
Confusion on licensing here. Not sure how to address, but this comment is super confusing. Does it override the licence for the overall library? Is this code dual licensed with both the library license and this licence here in the comment? I suggest to remove this comment, and somehow come to terms with the fact that you are re-licensing this code/algorithm by submitting this PR.

The original license does not apply to a JavaScript port, I don't think.